lib/rspec/matchers/built_in/raise_error.rb

```diff
@@ -46,7 +46,6 @@ def matches?(given_proc, negative_expectation=false, &block)
           @eval_block = false
           @eval_block_passed = false
 
-          warn_about_bare_error if warning_about_bare_error && !negative_expectation
           return false unless Proc === given_proc
```

Comment:
I disagree -- with composable matchers in RSpec 3 a matcher needs to be able to be asked if it matches against _any_ type of object, and it should signal it does or does not by returning true/false from `matches?`.  We should only raise errors from `matches?` when the matcher itself is invalid and cannot be matched against any object (such as when the user has done `be_within(delta)` without chaining `of(value)`).